server/lib/bdbag/bdbag_provider.py

```diff
@@ -54,23 +65,40 @@ def _add(self, path: pathlib.Path, url: Optional[str], orig_path: pathlib.Path,
 
 
 class BDBagProvider(ImportProvider):
-    def __init__(self) -> None:
-        super().__init__('BDBag')
+    def __init__(self, name: str = 'BDBag') -> None:
+        super().__init__(name)
 
     def matches(self, entity: Entity) -> bool:
-        return False
+        try:
+            return str(entity.getValue()).endswith('.zip')
+        except:
+            return False
 
     def _listRecursive(self, user: Dict[str, object], pid: str, name: str,
                        base_url: Optional[str] = None,
                        progress: Optional[object] = None) -> Generator[ImportItem, None, None]:
         # base_url + '/' + name is expected to be a path to a zip file
         if not pid:
             raise ValueError('pid must contain a path to a bag.')
-        with zipfile.ZipFile(pid) as zf:
-            with tempfile.TemporaryDirectory() as tmp_dir:
-                zf.extractall(tmp_dir)
+        if base_url:
+            pid = base_url + '/' + pid
```

Comment:
`base_url` is a bait. This thing exists solely for DataONE integration and should never be used by anything else.
```suggestion
```

Reply:
Well, yes and no. The pattern is there in multiple places. Deriva has multiple deployments. So does/did the Globus thing.

Reply:
That's ok. All I say is that `base_url` is used throughout our only client as something DataONE related, so it won't have any DERIVA specific values ever. AFAICT `pid` in this context is already a full blown url.

Reply:
It probably won't matter. The reason I'd keep it because, even though it's not used in practice, it makes sense in theory.